mk/spksrc.tc.mk

```diff
@@ -38,51 +39,44 @@ include ../../mk/spksrc.extract.mk
 patch: extract
 include ../../mk/spksrc.patch.mk
 
-fix: patch
-include ../../mk/spksrc.tc-fix.mk
-
-
-all: fix
-
+flags: patch
+include ../../mk/spksrc.tc-vers.mk
 
-TOOLS = ld ldshared:"gcc -shared" cpp nm cc:gcc as ranlib cxx:g++ ar strip objdump readelf
+vers: flags
+include ../../mk/spksrc.tc-flags.mk
 
-CFLAGS += $(TC_CFLAGS)
-CFLAGS += -I$(INSTALL_DIR)/$(INSTALL_PREFIX)/include
-
-CPPFLAGS += $(TC_CPPFLAGS)
-CPPFLAGS += -I$(INSTALL_DIR)/$(INSTALL_PREFIX)/include
-
-CXXFLAGS += $(TC_CXXFLAGS)
-CXXFLAGS += -I$(INSTALL_DIR)/$(INSTALL_PREFIX)/include
+fix: vers
+include ../../mk/spksrc.tc-fix.mk
 
-LDFLAGS += $(TC_LDFLAGS)
-LDFLAGS += -L$(INSTALL_DIR)/$(INSTALL_PREFIX)/lib
-LDFLAGS += -Wl,--rpath-link,$(INSTALL_DIR)/$(INSTALL_PREFIX)/lib
-LDFLAGS += -Wl,--rpath,$(INSTALL_PREFIX)/lib
+all: $(TC_LOCAL_VARS_MK)
 
+.PHONY: $(TC_LOCAL_VARS_MK)
+$(TC_LOCAL_VARS_MK): fix
+	env $(MAKE) --no-print-directory tc_vars > $@ 2>/dev/null;
 
 .PHONY: tc_vars
-tc_vars: patch
+tc_vars: fix
 	@echo TC_ENV :=
 	@for tool in $(TOOLS) ; \
 	do \
 	  target=`echo $${tool} | sed 's/\(.*\):\(.*\)/\1/'` ; \
 	  source=`echo $${tool} | sed 's/\(.*\):\(.*\)/\2/'` ; \
-	  echo TC_ENV += `echo $${target} | tr [:lower:] [:upper:] `=\"$(WORK_DIR)/$(TC_BASE_DIR)/bin/$(TC_PREFIX)-$${source}\" ; \
+	  echo TC_ENV += `echo $${target} | tr [:lower:] [:upper:] `=\"$(WORK_DIR)/$(TC_PREFIX)/bin/$(TC_PREFIX)-$${source}\" ; \
 	done
 	@echo TC_ENV += CFLAGS=\"$(CFLAGS) $$\(ADDITIONAL_CFLAGS\)\"
 	@echo TC_ENV += CPPFLAGS=\"$(CPPFLAGS) $$\(ADDITIONAL_CPPFLAGS\)\"
 	@echo TC_ENV += CXXFLAGS=\"$(CXXFLAGS) $$\(ADDITIONAL_CXXFLAGS\)\"
 	@echo TC_ENV += LDFLAGS=\"$(LDFLAGS) $$\(ADDITIONAL_LDFLAGS\)\"
-	@echo TC_CONFIGURE_ARGS := --host=$(TC_TARGET) --build=i686-pc-linux
-	@echo TC_TARGET := $(TC_TARGET)
-	@echo TC_PREFIX := $(TC_PREFIX)-
-	@echo TC_PATH := $(WORK_DIR)/$(TC_BASE_DIR)/bin/
+	@echo TC_CONFIGURE_ARGS := --host=$(TC_PREFIX) --build=i686-pc-linux
+	@echo TC_PREFIX := $(TC_PREFIX)
+	@echo TC_PATH := $(WORK_DIR)/$(TC_PREFIX)/bin/
```

Comment:
Does TC_PREFIX changes "meaning" from that point?

Reply:
That the whole point... I tried to reduce the amount of variable duplication to the maximum and might have stretched `TC_PREFIX` a bit too much.  I'll readjust.

Reply:
As mentioned on comment above: re-declared `TC_TARGET` and reverted `TC_PREFIX` to original intended behaviour.